src/cls/queue/cls_queue_src.cc

```diff
@@ -27,6 +27,8 @@ int queue_write_head(cls_method_context_t hctx, cls_queue_head& head)
 
   bl.claim_append(bl_head);
 
+  ceph_assert(bl.length() <= head.max_head_size);
```

Comment:
please return an error instead of assert.
you can assert at gc level, as it is a "software bug" there

Reply:
Ok

Reply:
agree, avoid inline asserts in general, esp. with the mess around assert in ceph